custom_components/adaptive_lighting/switch.py

```diff
@@ -830,6 +832,33 @@ async def _adapt_light(
             max_kelvin = attributes["max_color_temp_kelvin"]
             color_temp_kelvin = self._settings["color_temp_kelvin"]
             color_temp_kelvin = max(min(color_temp_kelvin, max_kelvin), min_kelvin)
+            if self._dim_to_warm and "brightness" in features:
+                # uncomment these lines for easy debugging using light.turn_on outside of the integration.
+                #await self.hass.helpers.entity_component.async_update_entity(light)
+                #cur_state = self.hass.states.get(light)
+                #brightness = cur_state.attributes[ATTR_BRIGHTNESS]
+                #service_data[ATTR_BRIGHTNESS] = brightness
+                min_ct = self._sun_light_settings.min_color_temp  # pylint: disable=protected-access
+                max_ct = color_temp_kelvin
+                max_brightness = self._sun_light_settings.max_brightness  # pylint: disable=protected-access
+                min_brightness = self._sun_light_settings.min_brightness  # pylint: disable=protected-access
+                max_brightness = max((max_brightness*2.55),brightness)
+                min_brightness = min((min_brightness*2.55),brightness)
+                # No max/min brightness attributes? Unfortunate.
+                #min_brightness = attributes["min_brightness"]
+                #max_brightness = attributes["max_brightness"]
+                _LOGGER.debug(
+                    "Setting color temp using the following values in eq: max_brightness: %s, min_brightness: %s, max_ct: %s, min_ct: %s, brightness: %s",
+                    max_brightness,
+                    min_brightness,
+                    max_ct,
+                    min_ct,
+                    brightness,
+                )
+                 # y = a(x-h)^2+k where h,k is the vertex (255,6500) or (max_brightness,max_ct)
+                 # a = (min_ct-max_ct)/(min_brightness-max_brightness)^2
+                 # check: y = (1000-6500)/((1-h)^2)*(x-255)^2+6500 if x=2 then y=1043.221836 when min_brightness=1,max_brightness(h)=255,max_ct=6500,min_ct=1000
+                color_temp_kelvin = ((min_ct-max_ct)/(min_brightness-max_brightness)**2)*(brightness-max_brightness)**2+max_ct
```

Comment:
What is the reasoning behind this equation?

Do you perhaps have a graph or something that we can add to the README?

Reply:
Yes! please look at the issue #78 I have a graph posted toward the bottom of the thread